### ISSUE TITLE
Simple Summary: Areas Alpaka Abstraction

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/alpaka#578]
We should summarize in the docs which areas Alpaka abstracts and which not.
This makes it easier for users to understand where to look for certain things when building on Alpaka.

Currently it does (feel free to edit) abstract:

- thread scheduling & indexing
- math primitives (abs, cos/sin/tan, exp, pow, ...)
- random number generators
- ...
- future: BLAS primitives in/ex kernel, e.g. [CUTLASS](https://github.com/NVIDIA/cutlass)-like?

It is not designed to abstract:
- ...

**Comment by HadrienG2** [comment]
"Task scheduling" indirectly covers device command submission (command queues, CUDA stream...), but it does not obviously cover device-host synchronization (events, wait for queue/device, etc), which Alpaka must also abstract. You will want to discuss that as well.

**Comment by sbastrakov** [comment]
I agree with @HadrienG2 that we should add device-host synchronization explicitly, as it seems indeed not covered by other items from the list.
